src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.cs

```diff
@@ -178,24 +178,30 @@ private static Glyph GetGlyph(SymbolKind kind, Accessibility? declaredAccessibil
 
             if (type is INamedTypeSymbol namedType && namedType.OriginalDefinition != null)
             {
-                var originalDefinition = namedType.OriginalDefinition;
-
-                var ienumerableOfT = namedType.GetAllInterfacesIncludingThis().FirstOrDefault(
-                    t => t.OriginalDefinition.SpecialType == SpecialType.System_Collections_Generic_IEnumerable_T);
+                var getEnumeratorMethod = namedType.GetMembers().OfType<IMethodSymbol>().FirstOrDefault(m => m.IsValidGetEnumerator() || m.IsValidGetAsyncEnumerator());
 
-                if (ienumerableOfT != null)
+                if (getEnumeratorMethod is not null)
                 {
+                    var members = getEnumeratorMethod.ReturnType.AllInterfaces.Concat(returnType.GetBaseTypesAndThis())
+                        .SelectMany(x => x.GetMembers())
+                        .Where(x => x.DeclaredAccessibility == Accessibility.Public)
+                        .ToList();
+
+                    var currentProperty = members.OfType<IPropertySymbol>().FirstOrDefault(p => p.Name == WellKnownMemberNames.CurrentPropertyName && p.GetMethod != null);
```

Comment:
A copy from:

https://github.com/dotnet/roslyn/blob/0ff47433e7fdb834f76fb711c5cdd78d0b0cb342/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ISymbolExtensions.cs#L709-L718


Probably figure a better approach to share the code instead of duplicating it?